### PR TITLE
Fix `make gen` problem

### LINKED ISSUE #571: `make gen` generate proto problem
**What happened**:

Run `make gen` locally produces some differences:
image

```
[root@localhost kmesh]# protoc --version
libprotoc 3.17.3
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kmesh version:
- Others:

### PR DESCRIPTION
**Pull request by Okabe-Rintarou-0**
**What type of PR is this?**
bug

**What this PR does / why we need it**:
Fix `make gen` problem, upgrate protoc-c to 1.4.x, aligning with the native one in openEuler 23.03.

**Which issue(s) this PR fixes**:
Fixes #571 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: hack/gen_protoc.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# if protoc-c is 1.4.x
+if protoc-c --version | grep -q -E "protobuf-c 1\.4\.[0-9]"; then
+    protoc-c --proto_path=$PROTO_PATH --c_out=. $PROTO_SRC
+else
+    echo "Generate proto files in docker"
+    docker run --rm \
+        --privileged=true \
+        -v $ROOT_DIR:/kmesh \
+        --name kmesh-build \
+        ghcr.io/kmesh-net/kmesh-build-${arch}:latest \
```
Comment (YaoZengzeng) [inline hack/gen_protoc.sh:21]: Does the build image already support multi arch? @lec-bit

File: hack/install-proto.sh [hunk 2]
```diff
@@ -3,6 +3,7 @@
 function install_tool () {
     tool=${1}
     if command -v apt > /dev/null; then
+        sudo apt-get update
```
Comment (hzxuzhonghu) [inline hack/install-proto.sh:6]: indents

File: hack/gen_protoc.sh (hunk 1 above)
Comment (hzxuzhonghu) [reply hack/gen_protoc.sh:21]: https://github.com/orgs/kmesh-net/packages/container/package/kmesh-build

We have supported both arm and x86， so can remove arch suffix

Comment (Okabe-Rintarou-0) [reply hack/gen_protoc.sh:21]: > https://github.com/orgs/kmesh-net/packages/container/package/kmesh-build
> 
> We have supported both arm and x86， so can remove arch suffix

then we should also remove the arch preffix in make build?

**Review by hzxuzhonghu — COMMENTED** [review]
/lgtm

/approve

**Pull request merged**